qutebrowser/commands/runners.py

```diff
@@ -285,6 +286,14 @@ def run(self, text, count=None):
             else:
                 result.cmd.run(self._win_id, args)
 
+            mode_manager = objreg.get('mode-manager', scope='window',
+                                      window=self._win_id)
+            if (result.cmdline[0] != 'repeat-command' and
+                    result.cmdline[0] != 'prompt-accept'):
```

Comment:
You can simplify this to `result.cmdline[0] not in ['repeat-command', 'prompt-accept']`. Thinking about it, I guess it makes sense to add `:leave-mode` as well as that's a mode-changing command (bound to `<esc>` usually) which you probably don't want to be repeated with `.` either.